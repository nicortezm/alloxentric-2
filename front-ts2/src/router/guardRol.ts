import Swal from "sweetalert2"




export const Role1Guard = ( to , from , next ) => {
    
    return new Promise ( () => {
     
        const rol = localStorage.getItem('rol')
  
        if( rol === 'role1' ) {

            console.log( 'Esta autorizado a filtrar')
            next()

        } else  {

            console.log( 'Bloqueado por el GuardRol' )

            next({ name: 'home'})

            Swal.fire({
                icon: 'warning',
                title: 'Rol invalido',
                text: `${rol} no tiene acceso a esa vista`,
              })
        }
    })
}



export const Role2Guard = ( to , from , next ) => {
    
    return new Promise ( () => {
     
        const rol = localStorage.getItem('rol')
  
        if( rol === 'role2' ) {

            console.log( 'Esta autorizado a etiquetar')
            next()

        } else  {

            console.log( 'Bloqueado por el GuardRol' )
            
            Swal.fire({
                icon: 'warning',
                title: 'Rol invalido',
                text: `${rol} no tiene acceso a esa vista`,
              })

            next({ name: 'home'})

        }
    })
}


