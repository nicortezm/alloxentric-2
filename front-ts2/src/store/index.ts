import { createStore } from 'vuex'


export default createStore({
  state: {
    user: {
      isAuthenticated: false,
      name: "",
      idToken: "",
      accessToken: "",
      rol: ""
    }
  },
  getters: {
  },


  mutations: {
    
    logout(state) {
      state.user.isAuthenticated = false;
      state.user.name = "";
      state.user.idToken = "";
      state.user.accessToken = "";
      state.user.rol = ""
    },

    login(state, payload) {
      state.user.isAuthenticated = true;
      state.user.idToken = payload.idToken;
      state.user.accessToken = payload.accessToken;
    },

    setName(state, payload) {
      state.user.name = payload.name;
    }
  },



  actions: {
  },



  modules: {
  }
})
