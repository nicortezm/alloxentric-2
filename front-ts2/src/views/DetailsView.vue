<template>

  <div class="row p-1">
      <div class="col d-flex justify-content-between">
         <div class="px-2">
            <a
               class="text-uppercase text-dark"
               @click="toFilter"
               style="cursor: pointer"
               ><i class="fa-solid fa-angle-left"></i> Volver</a
            >
        </div>
        <div class="px-2" v-if="isAuthenticated == true">
            <a class="text-dark mr-3">Bienvenido: <span class="text-uppercase">{{ getUsername() }}</span></a>
            <br>
            <button class="btn btn-secondary mt-2 text-uppercase" @click="onLogout"><i class="fa-solid fa-arrow-left"></i> Logout</button>
        </div>   
   </div>
  </div>


  <div class="row d-flex justify-content-center mt-3">
    <div class="col-lg-6 col-sm-12 mt-2">
      <table class="table" >
        <thead>
          <tr>
            <th scope="col">Audio Name</th>
            <th scope="col">{{ endpoint }}</th>
            <th scope="col">Percentage of Audio</th>
          </tr>
        </thead>

        <tbody v-for="val in values" :key="val.id" >
          <tr id="val">
            <td>{{ val.audio_name }}</td>
            <td>{{ val.value }}</td>
            <td>{{ val.percentage }}</td>
          </tr>
        </tbody>
       
      </table>     
    </div>
   
  </div>

  <div class="row d-flex justify-content-center mt-3">
      <button id="boton" @click="getValues()" class="btn btn-secondary">
         <i class="fa-solid fa-magnifying-glass"></i> Obtener Valores
      </button>
  </div>
</template>



<script lang="ts">

import FilterAPI from "../api/FilterApi";
export default {
  name: "DetailsView",

  data() {
      return {
         values: null,
      };
  },

  props: {
      endpoint: {
         type: String,
         required: true,
      },
  },

  computed:{
    isAuthenticated(){
      return this.$store.state.user.isAuthenticated;
    }
  },

  methods: {

      async getValues() {
        const resp = await FilterAPI.get(`/${this.$props.endpoint}`);
          
          console.log( 'hola' )
          this.values = resp.data;

          console.log( this.values )

        },

      toFilter() {
         this.$router.push({ name: "filter" });
      },
      onLogout(){
        let payloadRefreshedTokens = {
          idToken: '',
          accessToken: ''
      }
        this.$store.commit("login", payloadRefreshedTokens)
        this.$store.commit("logout")
    },
      getUsername(){
        return this.$store.state.user.name
      }
  },
};
</script>


<style>

</style>