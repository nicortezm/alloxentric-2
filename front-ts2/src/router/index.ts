import { createRouter, createWebHashHistory, RouteRecordRaw } from 'vue-router'
import HomeView from '../views/HomeView.vue'
import { Role1Guard, Role2Guard } from './guardRol';
import Keycloak from 'keycloak-js';




export const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'home',
    component: HomeView,
    props: Keycloak
  },
  {
    path: '/about',
    name: 'about',
    component: () => import(/* webpackChunkName: "about" */ '../views/AboutView.vue')
  },
  {
    path: '/filter',
    name: 'filter',
    beforeEnter: [ Role1Guard ], 
    component: () => import(/* webpackChunkName: "Filters" */ '../views/FilterView.vue'),
  },
  {
    path: '/tags',
    name: 'tags',
    beforeEnter: [ Role2Guard ], 
    component: () => import(/* webpackChunkName: "Tags" */ '../views/TagsView.vue')
  },
  {
    path: '/details/:endpoint',
    name: 'details',
    component: () => import(/* webpackChunkName: "Tags" */ '../views/DetailsView.vue'),
    props: ( route ) => {
      console.log( route )
      return {
          endpoint: route.params.endpoint
      }
    } 
  },

  { 
    path: '/:pathMatch(.*)*', 
    component: HomeView
  }    
]

const router = createRouter({
  history: createWebHashHistory(),
  routes,
})










export default router
