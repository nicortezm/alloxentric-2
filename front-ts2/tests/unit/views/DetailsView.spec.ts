// ts-nocheck ignores all ts errors in the file
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck

import DetailsView from '@/views/DetailsView.vue'

import { routes } from "@/router";
import { storeRole1, storeRole2 } from "../mockStore/store";
import { flushPromises, MountingOptions, shallowMount } from "@vue/test-utils";
import { createRouter, createWebHistory } from 'vue-router'

// @ts-ignore`
describe('Pruebas en la Vista DetailsView', () => {

   const mockRouter = {
      push: jest.fn(),
   }

   test('Debe de renderizar correctamente la vista', () => {
      
      const wrapper = shallowMount( DetailsView, {
         global: {
            plugins: [ storeRole1 ],
            mocks: {
               $router: mockRouter,           
            }
         },
         computed: {
            isAuthenticated() {
               return storeRole1.state.user.isAuthenticated
            }
         }, 
         props: {
            endpoint: 'silencio_total'
         }
      });

      expect( wrapper.html() ).toMatchSnapshot();
   });



   test('Al clickear boton obtener Valores la tabla se pobla', () => {

      const values = [
         { audio_duration:1, audio_name: 'test1.mp3', value: 1, percentage: 10 },
         { audio_duration:2, audio_name: 'test2.mp3', value: 2, percentage: 20 },
         { audio_duration:3, audio_name: 'test3.mp3', value: 3, percentage: 30 },
         { audio_duration:4, audio_name: 'test4.mp3', value: 4, percentage: 40 },
         { audio_duration:5, audio_name: 'test5.mp3', value: 5, percentage: 50 }
      ]

      const wrapper = shallowMount( DetailsView, {
         data() {
            return {
               values: [
                  { audio_duration:1, audio_name: 'test1.mp3', value: 1, percentage: 10 },
                  { audio_duration:2, audio_name: 'test2.mp3', value: 2, percentage: 20 },
                  { audio_duration:3, audio_name: 'test3.mp3', value: 3, percentage: 30 },
                  { audio_duration:4, audio_name: 'test4.mp3', value: 4, percentage: 40 },
                  { audio_duration:5, audio_name: 'test5.mp3', value: 5, percentage: 50 }
               ]
            }
         },
         global: {
            plugins: [ storeRole1 ],
            mocks: {
               $router: mockRouter,           
            }
         },
         computed: {
            isAuthenticated() {
               return storeRole1.state.user.isAuthenticated
            }
         }, 
         props: {
            endpoint: 'silencio_total'
         }
      });

      wrapper.find('#boton').trigger('click');

       expect( wrapper.find('table').text() )
      .toBe(`Audio Namesilencio_totalPercentage of Audio${ values[0].audio_name }${ values[0].value }${ values[0].percentage }${ values[1].audio_name }${ values[1].value }${ values[1].percentage }${ values[2].audio_name }${ values[2].value }${ values[2].percentage }${ values[3].audio_name }${ values[3].value }${ values[3].percentage }${ values[4].audio_name }${ values[4].value }${ values[4].percentage }`)
   });








});