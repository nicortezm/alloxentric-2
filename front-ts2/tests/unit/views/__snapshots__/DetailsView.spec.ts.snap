// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la Vista DetailsView Debe de renderizar correctamente la vista 1`] = `
<div class="row p-1">
  <div class="col d-flex justify-content-between">
    <div class="px-2"><a class="text-uppercase text-dark" style="cursor: pointer;"><i class="fa-solid fa-angle-left"></i> Volver</a></div>
    <div class="px-2"><a class="text-dark mr-3">Bienvenido: <span class="text-uppercase">jorge</span></a><br><button class="btn btn-secondary mt-2 text-uppercase"><i class="fa-solid fa-arrow-left"></i> Logout</button></div>
  </div>
</div>
<div class="row d-flex justify-content-center mt-3">
  <div class="col-lg-6 col-sm-12 mt-2">
    <table class="table">
      <thead>
        <tr>
          <th scope="col">Audio Name</th>
          <th scope="col">silencio_total</th>
          <th scope="col">Percentage of Audio</th>
        </tr>
      </thead>
    </table>
  </div>
</div>
<div class="row d-flex justify-content-center mt-3"><button id="boton" class="btn btn-secondary"><i class="fa-solid fa-magnifying-glass"></i> Obtener Valores </button></div>
`;
